src/System.Runtime/tests/System/GCTests.cs

```diff
@@ -157,14 +169,19 @@ public static void KeepAlive_Null()
 
         private class KeepAliveNullTest
         {
-            public static void Run()
+            [System.Runtime.CompilerServices.MethodImplAttribute(System.Runtime.CompilerServices.MethodImplOptions.NoInlining)]
+            private static void MakeAndNull()
             {
                 var obj = new TestObject();
                 obj = null;
+            }
+
+            public static void Run()
+            {
+                MakeAndNull();
 
                 GC.Collect();
                 GC.WaitForPendingFinalizers();
-                GC.KeepAlive(obj);
 
```

Comment:
since I never talked to the person who wrote the test I can't say for sure what the intent was. but the code seems to want to demonstrate that "passing in null to KeepAlive is not going to keep anything alive"....if obj = null wasn't there, KeepAlive would be to say "keep obj alive till this point" but since it was already assigned null, KeepAlive would just be called on null. of course if you run the test in debug build it's a different story as jit will just extend the lifetime anyway (whether KeepAlive is there or not....at least on desktop AFAIK).